cylc/flow/scripts/workflow_state.py

```diff
@@ -18,252 +18,409 @@
+
+    parser.add_option(
+        "--pretty",
+        help="Pretty-print outputs (the default is single-line output).",
+        action="store_true", dest="pretty_print", default=False)
+
+    parser.add_option(
+        "--old-format",
+        help="Print results in legacy comma-separated format.",
+        action="store_true", dest="old_format", default=False)
+
+    # Back-compat support for pre-8.3.0 command line options.
+    parser.add_option(
+        "-t", "--task", help=f"Task name. {OPT_DEPR_MSG}.",
+        metavar="NAME",
+        action="store", dest="depr_task", default=None)
+
+    parser.add_option(
+        "-p", "--point",
+        metavar="CYCLE",
+        help=f"Cycle point. {OPT_DEPR_MSG}.",
+        action="store", dest="depr_point", default=None)
+
+    parser.add_option(
+        "-T", "--task-point",
+        help="In task job scripts, task cycle point from the environment"
+             " (i.e., --point=$CYLC_TASK_CYCLE_POINT)",
+        action="store_true", dest="use_task_point", default=False)
 
     parser.add_option(
         "-S", "--status",
-        help="Specify a particular status or triggering condition to "
-             f"check for. {conds}{states}",
-        action="store", dest="status", default=None)
+        metavar="STATUS",
+        help=f"Task status. {OPT_DEPR_MSG1}.",
+        action="store", dest="depr_status", default=None)
 
+    # Prior to 8.3.0 --output was just an alias for --message
     parser.add_option(
         "-O", "--output", "-m", "--message",
-        help="Check custom task output by message string or trigger string.",
-        action="store", dest="msg", default=None)
-
-    WorkflowPoller.add_to_cmd_options(parser)
+        metavar="MSG",
+        help=f"Task output message. {OPT_DEPR_MSG2}.",
+        action="store", dest="depr_msg", default=None)
+
+    WorkflowPoller.add_to_cmd_options(
+        parser,
+        d_interval=INTERVAL,
+        d_max_polls=MAX_POLLS
+    )
 
     return parser
 
 
 @cli_function(get_option_parser, remove_opts=["--db"])
-def main(parser: COP, options: 'Values', workflow_id: str) -> None:
-
-    if options.use_task_point and options.cycle:
-        raise InputError(
-            "cannot specify a cycle point and use environment variable")
-
-    if options.use_task_point:
-        if "CYLC_TASK_CYCLE_POINT" not in os.environ:
-            raise InputError("CYLC_TASK_CYCLE_POINT is not defined")
-        options.cycle = os.environ["CYLC_TASK_CYCLE_POINT"]
-
-    if options.offset and not options.cycle:
-        raise InputError(
-            "You must target a cycle point to use an offset")
-
-    # Attempt to apply specified offset to the targeted cycle
-    if options.offset:
-        options.cycle = str(add_offset(options.cycle, options.offset))
-
-    # Exit if both task state and message are to being polled
-    if options.status and options.msg:
-        raise InputError("cannot poll both status and custom output")
```

Comment:
This check has been lost

Reply:
Ah, yeah, so I originally removed all of these options, then ended up restoring them for CLI back-compat, to avoid breaking existing hand-rolled polling tasks.  I'll restore the check...